src/integration_tests/schemas/e2e_schema_integration_test.py

```diff
@@ -12,122 +12,173 @@
+        for survey_id in test_survey_id_list:
+            if survey_id in E2ESchemaIntegrationTest.schema_metadatas_dict:
+                continue # Skip if schema metadata already retrieved
+            schema_metadata_response = self.session.get(
+                f"{config.API_URL}/v1/schema_metadata?survey_id={survey_id}",
+                headers=self.headers,
+            )
+            assert schema_metadata_response.status_code == 200
+            # Add json to dict with survey_id as key
+            E2ESchemaIntegrationTest.schema_metadatas_dict[survey_id] = schema_metadata_response.json()
+            assert len(E2ESchemaIntegrationTest.schema_metadatas_dict[survey_id]) > 0
+        
+            for schema_metadata in E2ESchemaIntegrationTest.schema_metadatas_dict[survey_id]:
+                assert schema_metadata == {
+                    "guid": schema_metadata["guid"],
+                    "survey_id": survey_id,
+                    "schema_location": f"{survey_id}/{schema_metadata['guid']}.json",
+                    "sds_schema_version": schema_metadata["sds_schema_version"],
+                    "sds_published_at": schema_metadata["sds_published_at"],
+                    "schema_version": self.test_schema["properties"]["schema_version"]["const"],
+                    "title": self.test_schema["title"],
+                }
+
+    @pytest.mark.order(3)
+    def test_get_schema_v1(self):
+        """
+        Test the GET /v1/schema endpoint by retrieving the schema both by version and latest version and checking the response.
 
-        # Post the schema and check the response
-        schema_post_response = session.post(
+        * We retrieve the schema by version and check the response
+        * We POST a different schema for the same survey_id to test the latest version retrieval
+        """
+        for survey_id in test_survey_id_list:
+            for schema_metadata in E2ESchemaIntegrationTest.schema_metadatas_dict[survey_id]:
+                # Verify schema retrieval by version
+                set_version_schema_response = self.session.get(
+                    f"{config.API_URL}/v1/schema?"
+                    f"survey_id={schema_metadata['survey_id']}&version={schema_metadata['sds_schema_version']}",
+                    headers=self.headers,
+                )
+
+                assert set_version_schema_response.status_code == 200
+                assert set_version_schema_response.json() == self.test_schema
+
+                # Verify schema retrieval by the latest version
+                latest_version_schema_response = self.session.get(
+                    f"{config.API_URL}/v1/schema?survey_id={schema_metadata['survey_id']}",
+                    headers=self.headers,
+                )
+
+                assert latest_version_schema_response.status_code == 200
+                assert latest_version_schema_response.json() == self.test_schema
```

Comment:
Same here, it can't validate if it is returning the right schema for us as they are the same schema